gcc/testsuite/rust/compile/inline_asm_typecheck.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(rustc_attrs)]
+
+#[rustc_builtin_macro]
+macro_rules! asm {
+    () => {};
+}
+
+fn main() {
+    let mut _num1: i32 = 10;
+    let mut _num2: i32 = 10;
+    unsafe {
+        // This demonstrates that asm!'s is inferred with a unit type is parsed correctly.
+        let _ = asm!("nop");
+
+        // This errors out per rust spec
+        //      The asm! block never returns, and its return type is defined as ! (never).
+        //      Behavior is undefined if execution falls through past the end of the asm code.
+        //      A noreturn asm block behaves just like a function which doesn't return; notably, local variables in scope are not dropped before it is invoked.
+        let _ = asm!("nop", options(noreturn));
```

Comment:
hi Arthur, I dont think this needs a test case in compile time since rust playground also compiles this fine with no warnings. It only exited with "Illegal instruction: 4" in execute.

I'm not sure why I wrote that this errors out in the comment... Sorry for the confusion. I'll remove that comment

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bea3af18a69e3153cd2e4c456251d5bb